### PR TITLE
Feature/1949

### LINKED ISSUE #1949: Implement /changelog API serving CHANGELOG file and implement Admin UI its visualization
Since we are planning on removing `username` in the mvp/multitenancy branch we need to handle the migration of the schema well and informing users that they now need to use their email address for logging in.

I am in favor of displaying a message on `/#/admin` and `/#/login`. I feel that admin that suddenly lose the ability to log in will cause a lot of unnecessary confusion. We can require user interaction in the migration that walks through each username + mail address combination with some messaging that explains what is happening.

Further in the admin interface it would be sane to display system messages on the first login of an admin into an updated system.

### PR DESCRIPTION
**Pull request by NSkelsey**
This PR closes #1949 without requiring backend changes or the usage of local storage to display the change set the user has already seen. We generally want the ability to notify the user of a set of events that have occurred since the last time they logged in. These change log entries are now one of those events.

Notably here the backend now exports `changelog.json` in `./data/`. This may be caught by future pen testers eager to point out that it is not a best practice, but after some disucssion with @evilaliv3 we are not convinced it is good practice to withhold the system version.

Finally please note the `resolve` usage in `app.js`. This form of promise injection is idiomatic and will be required in the move to angular v2. Note that the changelog list is injected into the scope as `$scope.$resolve.changelog`

### CODE REVIEW

**Comment by NSkelsey** [comment]
A screen of the interesting part:
admin-home

**Comment by NSkelsey** [comment]
@evilaliv3 review please.

File: CHANGELOG [hunk 1]
```diff
@@ -13,7 +13,7 @@ Changes in version 2.76.2 - 2017-04-01
 Changes in version 2.67.1 - 2017-03-17
     o Start (restart) apparmor upon globaleaks install
 
-Changes in version 2.67.0 - 2017-03-17
+Changes in version 2.67.0 - 2017-03-17 - alert:notice
```
Comment (evilaliv3) [inline CHANGELOG]: this has been forgotten here while merging.

going to remove it.

File: client/Gruntfile.js [hunk 2]
```diff
@@ -782,15 +782,55 @@ module.exports = function(grunt) {
...
+          }
+          chnglog.push(obj);
+        } else {
+          obj.txt = obj.txt + '\n' + l;
+        }
+      }
+      // Intentionally drop the first release because does not conform to the format.
+
+      // Using object here due to issues with exporting raw lists via json
+      var output = JSON.stringify({'v': chnglog});
```
Comment (evilaliv3) [inline client/Gruntfile.js]: in general javascript variables should be declared on top.

File: client/app/views/admin/about/changelog.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<h3>System updates</h3>
```
Comment (evilaliv3) [inline client/app/views/admin/about/changelog.html]: missing data-translate, going to add it

File: client/app/views/admin/about/releasemsg.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<div>
+  <div class="info small">
+    <span data-ng-show="::change.alert"><span class="glyphicon glyphicon-exclamation-sign text-danger"></span>
+      <span data-translate>Major changes</span>
+    </span>
+    <span data-ng-hide="::change.alert"><span class="glyphicon glyphicon-plus-sign text-muted"></span>
+      <span data-translate>Minor changes</span>
+    </span>
+  </div>
+  <pre style="width: 85ch; white-space: pre-line;">{{::change.txt }}</pre>
```
Comment (evilaliv3) [inline client/app/views/admin/about/releasemsg.html]: let's avoid inline css

File: client/app/views/admin/landing.html [hunk 5]
```diff
@@ -2,40 +2,54 @@
...
+          <uib-tab data-index="0" heading="{{'Home' | translate}}">
+            <div class="tab-content" data-ng-include="'views/admin/about/contact.html'"></div>
+          </uib-tab>
+
+          <uib-tab data-index="1">
+            <uib-tab-heading>
+              <span data-translate>Change log</span>
+            </uib-tab-heading>
+            <div class="tab-content">
+              <div class="btn-group" role="group" aria-label="Release notes toggle.">
```
Comment (evilaliv3) [inline client/app/views/admin/landing.html]: lets avoid for the moment put aria labels and

Comment (evilaliv3) [inline client/app/views/admin/landing.html]: for uniformity reasons let's avoid to start aria labels until propre analisys of what to do in relations to manual aria settings.

File: client/Gruntfile.js [hunk 6]
```diff
@@ -782,15 +782,55 @@ module.exports = function(grunt) {
...
+
+      cnt = grunt.file.read('../CHANGELOG');
+      var lines = cnt.split("\n"),
+          chnglog = [],
+          obj;
+
+      for (var i = 0; i < lines.length; i++) {
+        var l = lines[i];
+        // Format of the first line of every new release is:
+        // Changes in version 2.76.4 - 2017-04-12 - alert:notice
```
Comment (evilaliv3) [inline client/Gruntfile.js]: nice implementatio and ideas @NSkelsey 

but as discussed a week ago on slack also with @fpietrosanti let's avoid for now to implement filters.

prioritiziong means decide not to show something; i would suggest to skip until proper understanding of how to categorize messagges.

**Comment by evilaliv3** [comment]
i'm sorry for having taken so long to review this pull @NSkelsey !

nice work!

**Pull request merged**